ui/src/core_components/embed/CorePDF.vue

```diff
@@ -33,26 +60,251 @@ export default {
 				type: FieldType.Text,
 				desc: "A valid URL. Alternatively, you can provide a state reference to a packed PDF file.",
 			},
+			highlights: {
+				name: "Highlights",
+				default: JSON.stringify([]),
+				desc: "A list of highlights to be applied to the PDF as a JSON array of strings.",
+				type: FieldType.Object,
+			},
+			selectedMatch: {
+				name: "Selected highlight match",
+				default: null,
+				desc: "The index of the selected highlight match.",
+				type: FieldType.Number,
+			},
+			page: {
+				name: "Page",
+				type: FieldType.Number,
+				desc: "The page to be displayed.",
+			},
 			cssClasses,
 		},
+		events: {
+			"pdf-load": {
```

Comment:
Is this event being fired? I can't seem to find the code that fires it.

Feel free to remove this event for now, but if you want to keep it:

- Make sure to trigger it
- Please rename this one to `ss-pdf-load`. You shouldn't need to change the core if there's no payload.
- Also, add a `desc` here even if we suspect what is, just for consistency.

**Note about "ss-" usage:**

I strongly prefer to prefix "ss-" and use sanitisation when needed, but when basically forwarding events from other libraries e.g. Google Maps this is potentially a pain as you usually have complex types that can change across versions.